tests/rkt_fetch_test.go

```diff
@@ -168,3 +174,180 @@ func testFetchNoStore(t *testing.T, args string, image string, finalURL string)
+			panic(err)
+		}
+		reportkill <- struct{}{}
+	}
+}
+
+func TestResumedFetchCacheUnsupported(t *testing.T) {
+	image := "rkt-inspect-implicit-fetch.aci"
+	imagePath := patchTestACI(image, "--exec=/inspect")
+
+	defer os.Remove(imagePath)
+
+	kill := make(chan struct{})
+	reportkill := make(chan struct{})
+
+	server := httptest.NewServer(testInterruptingServerHandler(t, imagePath, kill, reportkill, true, true))
+	defer server.Close()
+
+	ctx := testutils.NewRktRunCtx()
+	defer ctx.Cleanup()
+
+	// rkt should fetch from the start of the image both times in this loop,
+	// since the Last-Modified header will be newer than the cached version. If
+	// rkt doesn't start the second fetch from the start of the image, the
+	// `<-kill` line will wait indefinitely.
+	cmd := fmt.Sprintf("%s --no-store --insecure-skip-verify fetch %s", ctx.Cmd(), server.URL)
+	for i := 0; i < 2; i++ {
+		child := spawnOrFail(t, cmd)
+		<-kill
+		err := child.Close()
+		if err != nil {
+			panic(err)
+		}
+		reportkill <- struct{}{}
+	}
+}
+
+func testInterruptingServerHandler(t *testing.T, imagePath string, kill, waitforkill chan struct{}, supportsLastMod bool, lastModTooNew bool) http.HandlerFunc {
+	finfo, err := os.Stat(imagePath)
+	if err != nil {
+		panic(err)
+	}
+	cutoff := finfo.Size() / 2
+	return func(w http.ResponseWriter, r *http.Request) {
+		if r.Method == "HEAD" {
+			headers := w.Header()
+			headers["Accept-Ranges"] = []string{"bytes"}
+			if supportsLastMod {
+				if lastModTooNew {
+					headers["Last-Modified"] = []string{time.Now().Add(time.Minute).Format("Mon, 02 Jan 2006 15:04:05 MST")}
+				} else {
+					headers["Last-Modified"] = []string{"Mon, 02 Jan 2006 15:04:05 MST"}
+				}
+			}
+			w.WriteHeader(http.StatusOK)
+		}
+		if r.Method != "GET" {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		file, err := os.Open(imagePath)
+		if err != nil {
+			panic(err)
+		}
+		defer file.Close()
+
+		rangeHeaders, ok := r.Header["Range"]
+		if ok && len(rangeHeaders) == 1 && strings.HasPrefix(rangeHeaders[0], "bytes=") {
+			rangeHeader := rangeHeaders[0][6:] // The first (and only) range header, with len("bytes=") characters chopped off the front
+			tokens := strings.Split(rangeHeader, "-")
+			if len(tokens) != 2 {
+				t.Fatalf("number of '-'s in range header was not 1, it was %d", len(tokens))
+			}
+
+			start, err := strconv.Atoi(tokens[0])
+			if err != nil {
+				if tokens[0] == "" {
+					start = 0 // If start wasn't specified, start at the beginning
+				} else {
+					t.Fatalf("requested non-int starting location: %s", tokens[0])
+				}
+			}
+			end, err := strconv.Atoi(tokens[1])
+			if err != nil {
+				if tokens[1] == "" {
+					end = int(finfo.Size()) // If end wasn't specified, end at the end
+				} else {
+					t.Fatalf("requested non-int ending location: %s", tokens[0])
+				}
+			}
+
+			_, err = file.Seek(int64(start), os.SEEK_SET)
+			if err != nil {
+				panic(err)
+			}
+
+			_, err = io.CopyN(w, file, int64(end-start))
```

Comment:
I'll need to add a `- 1` to the `end = int(finfo.Size())` a few lines up then. I'd prefer to use `CopyN`, because while it won't be used in this test it wouldn't behave correctly if an end value was specified.

For verifying the hash, sure.